src/providers/authentication/__tests__/authentication.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { TestBed } from '@angular/core/testing';
+import { InAppBrowser } from '@ionic-native/in-app-browser';
+import { MSAdal } from '@ionic-native/ms-adal';
+
+import { MSAdalMock } from '../__mocks__/ms-adal.mock';
+import { AuthenticationProvider } from '../authentication';
+import { AppConfigProvider } from '../../app-config/app-config';
+import { AppConfigProviderMock } from '../../app-config/__mocks__/app-config.mock';
+import { InAppBrowserMock } from '../__mocks__/in-app-browser.mock';
+
+describe('Authentication', () => {
+  let authenticationProvider: AuthenticationProvider;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        AuthenticationProvider,
+        { provide: MSAdal, useClass: MSAdalMock },
+        { provide: AppConfigProvider, useClass: AppConfigProviderMock },
+        { provide: InAppBrowser, useClass: InAppBrowserMock}
+      ]
+    })
+  });
+
+  beforeEach(() => {
+    authenticationProvider = TestBed.get(AuthenticationProvider)
+  });
+
+  describe('Provider', () => {
+
+    it('should compile', () => {
+      expect(authenticationProvider).toBeDefined;
+    });
+
+    it('isAuthenticated() should return false when no login has happened', () => {
+      expect(authenticationProvider.isAuthenticated()).toEqual(false);
+    });
+
+    it('getAuthenticationToken() should return undefined if no login has happened', () => {
+      expect(authenticationProvider.getAuthenticationToken()).toEqual(undefined);
+    })
+
+    it('should silently login successfully', async () => {
+      await authenticationProvider.login();
+
+      expect(authenticationProvider.isAuthenticated()).toEqual(true);
+      expect(authenticationProvider.getAuthenticationToken()).toEqual('SILENT AYSNC TEST TOKEN')
+    });
+
+    it('should sign in with credetials', async() => {
+      await authenticationProvider.loginWithCredentials();
+
+      expect(authenticationProvider.isAuthenticated()).toEqual(true);
+      expect(authenticationProvider.getAuthenticationToken()).toEqual('AYSNC TEST TOKEN');
+    });
+
+    it('should do something when login fails', async () => {
+      // TODO
```

Comment:
Seems quite an important test to elaborate.

Reply:
The function in the authentication provide it doesn't do anything either, I guess we can use it for logging or retries if we decide to implement them. It didn't seem right to restrict access/redirect people from here (I think it's better the pages control that) - The other option was to remove that test but I think it works as a reminder this is something we need to look at in the future